### ISSUE TITLE
Async command Error: stderr maxBuffer exceeded

### ISSUE THREAD

**Issue by bmcallis** [pawelgalazka/runjs#43]
I'm trying to use runjs for our webpack build. When I run it with the async option I get the following error `stderr maxBuffer exceeded`

Here is my runfile.js
```
import { run } from 'runjs';
export default {
    build: (options={}) => {
        run('webpack --progress', {async: true});
    },
};
```

And the full output from running the command
```
$ run build
Requiring babel-register...
Processing runfile...
Running "build"...
webpack --progress
Finished "build" in 0.01 sec
 38% building modules 373/794 modules 421 active .../path/to/some/file.js 
(node:72847) UnhandledPromiseRejectionWarning: Unhandled promise rejection 
(rejection id: 1): Error: stderr maxBuffer exceeded
```

**Comment by bmcallis** [comment]
I've dug into this a bit more and the buffer won't be exceed if the code is switched from `childProcess.exec` to `childProcess.spawn`. However they are called differently so the api would have to changed. Here is a hardcoded runAsync function using spawn.

```
function runAsync (command, options) {
  // Prepare options for exec command (don't need async and stdio as it doesn't handle them)
  const execOptions = Object.assign({}, options)
  delete execOptions.async
  delete execOptions.stdio

  return new Promise((resolve, reject) => {
    const asyncProcess = childProcess.spawn('webpack', ['--progress'], execOptions)
      asyncProcess.on('close', code => {
        if (code !== 0) {
          reject(`Command failed with code: ${code}`)
        } else {
          resolve()
        }
      });

    // Simulate stdio=inherit behaviour for exec async (exec doesn't handle stdio option)
    if (options.stdio === 'inherit') {
      asyncProcess.stdout.pipe(process.stdout)
      asyncProcess.stderr.pipe(process.stderr)
    }
  })
}
```

**Comment by pawelgalazka** [comment]
Thanks for posting. Yeah I also encounter this issue with webpack but rarely. Anyway it's a bug. Feel free to open PR if you feel like it.

**Comment by pawelgalazka** [comment]
Fixed by #45
